src/GovUk.Education.ExploreEducationStatistics.Common/Config/SensitiveDataTelemetryProcessor.cs

```diff
@@ -0,0 +1,160 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Web;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Config
+{
+    /// <summary>
+    /// Tries to filter out any potentially sensitive data from telemetry
+    /// before it is sent to Application Insights e.g. query string parameters
+    /// containing things like access tokens or codes.
+    /// </summary>
+    public class SensitiveDataTelemetryProcessor : ITelemetryProcessor
+    {
+        private const string RedactedValue = "__redacted__";
+
+        private readonly HashSet<string> _redactedTokens = new()
+        {
+            "code",
```

Comment:
Was wondering if there was some OWASP list of common vulnerable tokens that was maintained?  Looks like a good starter for ten here.  Can't immediately think of any more to add off the top of my head.

Reply:
Doesn't seem to be, but I've added a few more around 'user' and 'name' like tokens. Some of these are a little overkill anyway. They mostly only help with cases where we've accidentally using query parameters in a non-secure way, but most of the time this should be caught in code review and be heavily discouraged (use request bodies instead).